code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1997,3 +1997,24 @@
 				var/scale = min(1, round(P.damage / 50, 0.2))
 				var/matrix/M = new()
 				B.transform = M.Scale(scale)
+
+/mob/living/carbon/human/apply_radiation_effects()
+	. = ..()
+	if(. == TRUE)
+		if(src.is_diona())
+			var/damage = rand(15, 30)
+			src.adjustToxLoss(-damage)
+			if(prob(5))
+				damage = rand(20, 60)
+				src.adjustToxLoss(-damage)
+			to_chat(src, "<span class='notice'>You can feel flow of energy which makes you regenerate.</span>")
```

Comment:
replace tbe <span class='notice'> and </span> with SPAN_NOTICE(), feel free to look at some of the other code to see how it works if ya want